### PR TITLE
Fix __virtual__ issue in mac_system.py

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes a race condition processing the `__virtual__` for `mac_system`. It was trying to execute a `__salt__['cmd.run']` before it was available in `__salt__`. Moved those functions to `salt.util.mac_utils`.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/45673

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/mac_service.py [hunk 1]
```diff
@@ -195,26 +127,7 @@ def launchctl(sub_cmd, *args, **kwargs):
...
-    # Raise an error or return successful result
-    if ret['retcode']:
-        out = 'Failed to {0} service:\n'.format(sub_cmd)
-        out += 'stdout: {0}\n'.format(ret['stdout'])
-        out += 'stderr: {0}\n'.format(ret['stderr'])
-        out += 'retcode: {0}\n'.format(ret['retcode'])
-        raise CommandExecutionError(out)
-    else:
-        return ret['stdout'] if return_stdout else True
+    return salt.utils.mac_utils.launchctl(sub_cmd, args, kwargs)
```
Comment (rallytime) [inline salt/modules/mac_service.py]: Should this be `**kwargs`?

**Review by rallytime — CHANGES_REQUESTED** [review]
Just one question here from me.

There should also be some unittests written against these new functions.

File: salt/modules/mac_service.py (hunk 1 above)
Comment (twangboy) [reply salt/modules/mac_service.py]: Probably

**Comment by twangboy** [comment]
@rallytime I've added the unit tests. They pass individually, but not when run together. I need help getting this figured out.

**Comment by rallytime** [comment]
@twangboy Were you able to come back to this change?

**Comment by twangboy** [comment]
@rallytime I need help with the tests. I'll fix the merge conflict

**Comment by twangboy** [comment]
@rallytime OK, I think we got this fixed. @terminalmage figured it out.

**Comment by rallytime** [comment]
@twangboy Great!

File: salt/utils/mac_utils.py [hunk 2]
```diff
@@ -227,3 +231,133 @@ def confirm_updated(value, check_fun, normalize_ret=False, wait=5):
...
+
+
+def _available_services():
+    '''
+    This is a helper function needed for testing. We are using the memoziation
+    decorator on the `available_services` function, which causes the function
+    to run once and then return the results of the first run on subsequent
+    calls. This causes problems when trying to test the functionality of the
+    `available_services` function. So this function is needed until we can
+    figure out how to properly mock the memoization decorator.
```
Comment (terminalmage) [inline salt/utils/mac_utils.py]: Mocking a decorator will cause you to enter a world of pain. I don't think we should bother attempting, TBH, so we can remove this last sentence from the docstring.

**Review by terminalmage — APPROVED** [review]
One small quibble in the docstring, but this otherwise ready to merge.

**Comment by rallytime** [comment]
@twangboy This also has a line failure: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/19750/violations/file/salt/utils/mac_utils.py/

**Comment by twangboy** [comment]
@rallytime Lint fixed... sentence removed...
Rebased

**Comment by rallytime** [comment]
re-run py3

**Comment by rallytime** [comment]
re-run py3

**Comment by rallytime** [comment]
@twangboy It looks like something isn't quite right on these tests for Python 3: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu16-py3/7733/

Failing tests:
- unit.utils.test_mac_utils.MacUtilsTestCase.test_available_services_non_xml
- unit.utils.test_mac_utils.MacUtilsTestCase.test_available_services_non_xml_malformed_plist

**Comment by cachedout** [comment]
@twangboy When will you be able to come back to this?

**Comment by twangboy** [comment]
@rallytime @cachedout Fixed tests for Py3 and rebased

**Comment by cachedout** [comment]
@rallytime Re-review from you needed then.

**Review by rallytime — APPROVED** [review]

**Pull request merged**